editoast/src/views/v2/path/pathfinding.rs

```diff
@@ -95,117 +94,189 @@ pub async fn post(
         infra_id: *infra_id,
     })
     .await?;
+
     Ok(Json(
-        pathfinding_blocks(conn, &mut redis_conn, core, &infra, &path_input).await?,
+        pathfinding_blocks(conn, &mut redis_conn, core, &infra, path_input).await?,
     ))
 }
 
+/// Pathfinding computation given a path input
 async fn pathfinding_blocks(
     conn: &mut DbConnection,
     redis_conn: &mut RedisConnection,
     core: Arc<CoreClient>,
     infra: &Infra,
-    path_input: &PathfindingInput,
+    path_input: PathfindingInput,
 ) -> Result<PathfindingResult> {
-    // Compute unique hash of PathInput
-    let hash = path_input_hash(infra.id, &infra.version, path_input);
+    let mut path = pathfinding_blocks_batch(conn, redis_conn, core, infra, &[path_input]).await?;
+    Ok(path.pop().unwrap())
+}
+
+/// Pathfinding batch computation given a list of path inputs
+async fn pathfinding_blocks_batch(
+    conn: &mut DbConnection,
+    redis_conn: &mut RedisConnection,
+    core: Arc<CoreClient>,
+    infra: &Infra,
+    path_inputs: &[PathfindingInput],
+) -> Result<Vec<PathfindingResult>> {
+    // Compute hashes of all path_inputs
+    let hashes: Vec<_> = path_inputs
+        .iter()
+        .map(|input| path_input_hash(infra.id, &infra.version, input))
+        .collect();
+
     // Try to retrieve the result from Redis
-    let result: Option<PathfindingResult> =
-        redis_conn.json_get_ex(&hash, CACHE_PATH_EXPIRATION).await?;
-    if let Some(pathfinding) = result {
-        info!("Hit cache");
-        return Ok(pathfinding);
-    }
-    // If miss cache:
+    let mut results: Vec<Option<PathfindingResult>> = redis_conn.json_get_bulk(&hashes).await?;
+
+    // Report number of hit cache
+    let nb_hit = results.iter().filter(|res| res.is_some()).count();
+    info!(nb_hit, nb_miss = path_inputs.len() - nb_hit, "Hit cache");
+
+    // Handle miss cache:
     // 1) extract locations from path items
-    let path_items = path_input.clone().path_items;
-    if path_items.len() <= 1 {
-        return Ok(PathfindingResult::NotEnoughPathItems);
+    let mut to_cache = vec![];
+    let mut pathfinding_requests = vec![];
+    let mut pathfinding_requests_index = vec![];
+    for (index, (result, path_input)) in results.iter_mut().zip(path_inputs).enumerate() {
+        if result.is_some() {
+            continue;
+        }
+        let path_items = path_input.clone().path_items;
```

Comment:
I don't believe you have to clone here.

```suggestion
        let path_items = &path_input.path_items;
```